### PR TITLE
feat: support dev rslib mf format assets

### PR DESCRIPTION
**Pull request by nyqykk**
Support use storybook to dev rslib mf format assets

### CODE REVIEW

File: packages/addon-rslib/src/init.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { createRsbuild, mergeRsbuildConfig } from '@rsbuild/core'
+import { unstable_composeCreateRsbuildConfig } from '@rslib/core'
+
+import type { RsbuildConfig, RsbuildInstance } from '@rsbuild/core'
+import type { RslibConfig } from '@rslib/core'
+
+export async function startMFDevServer(
```
Comment (fi3ework) [inline packages/addon-rslib/src/init.ts:7]:
```suggestion
export async function startMfDevServer(
```
as `mf` in `mfRsbuildConfig` is not capitalized.

File: sandboxes/rslib-react-component/stories/mfCounter.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// @ts-ignore
```
Comment (fi3ework) [inline sandboxes/rslib-react-component/stories/mfCounter.stories.tsx:1]: Is it common to add `@ts-ignore` for remote modules?

File: sandboxes/rslib-react-component/src/components/Counter/index.tsx [hunk 3]
```diff
@@ -1,7 +1,6 @@
 import React from 'react'
 import { CounterButton } from '../CounterButton/index'
 import { useCounter } from '../hooks/useCounter'
-import './index.scss'
```
Comment (fi3ework) [inline sandboxes/rslib-react-component/src/components/Counter/index.tsx:4]: Why it's removed here

File: sandboxes/rslib-react-component/tsconfig.json [hunk 4]
```diff
@@ -4,7 +4,10 @@
     "strict": true,
     "resolveJsonModule": true,
     "jsx": "react",
-    "outDir": "dist"
+    "outDir": "dist",
+    "paths": {
+      "*": ["./@mf-types/*"]
```
Comment (fi3ework) [inline sandboxes/rslib-react-component/tsconfig.json:9]: Is any code importing files like `import abd from '*/abc.js'`?

**Review by fi3ework — COMMENTED** [review]
The documentation of addon-rslib need to be updated alongside.

File: packages/addon-rslib/src/init.ts (hunk 1 above)
Comment (nyqykk) [reply packages/addon-rslib/src/init.ts:7]: this either all capital letters or all small letters, so better to keep captital~

File: sandboxes/rslib-react-component/stories/mfCounter.stories.tsx (hunk 2 above)
Comment (nyqykk) [reply sandboxes/rslib-react-component/stories/mfCounter.stories.tsx]: i write the reason in note

File: sandboxes/rslib-react-component/tsconfig.json (hunk 4 above)
Comment (nyqykk) [reply sandboxes/rslib-react-component/tsconfig.json:9]: no, this is for remote module types. https://module-federation.io/guide/start/quick-start.html#3-consuming-the-producer

Comment (fi3ework) [reply sandboxes/rslib-react-component/tsconfig.json:9]: magic.

File: packages/addon-rslib/src/init.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  if (process.env.NODE_ENV === 'production') {
+    // storybook build should skip this.
+    return
+  }
+  const rsbuildInstance = await initMFRsbuild(config)
+  if (!rsbuildInstance) {
+    // no mf format, return.
+    return
+  }
+  await rsbuildInstance.startDevServer()
```
Comment (fi3ework) [inline packages/addon-rslib/src/init.ts:19]: what if we move this line into `initMFRsbuild` to make the MF functionality more more cohesive.

**Review by fi3ework — COMMENTED** [review]
Generally LGTM.

**Review by fi3ework — APPROVED** [review]
Great work, I'll polish docs afterward.

**Pull request merged**